patterns/vpc-lattice/01_client-server-communication/main.tf

```diff
@@ -0,0 +1,396 @@
+    }
+  }
+}
+
+locals {
+  name   = basename(path.cwd)
+  region = "us-west-2"
+
+  cluster_vpc_cidr = "10.0.0.0/16"
+  client_vpc_cidr  = "10.1.0.0/16"
+  azs              = slice(data.aws_availability_zones.available.names, 0, 3)
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/aws-ia/terraform-aws-eks-blueprints"
+  }
+}
+
+################################################################################
+# Cluster
+################################################################################
+
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "~> 19.21"
+
+  cluster_name                   = local.name
+  cluster_version                = "1.28"
+  cluster_endpoint_public_access = true
+  enable_irsa                    = true
+
+  vpc_id     = module.cluster_vpc.vpc_id
+  subnet_ids = module.cluster_vpc.private_subnets
+
+  eks_managed_node_groups = {
+    initial = {
+      instance_types = ["m5.large"]
+
+      min_size     = 3
+      max_size     = 10
+      desired_size = 3
+    }
+  }
+
+  tags = local.tags
+}
+
+################################################################################
+# Cluster VPC
+################################################################################
+
+module "cluster_vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 5.4"
+
+  name = local.name
+  cidr = local.cluster_vpc_cidr
+
+  azs             = local.azs
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.cluster_vpc_cidr, 4, k)]
+  public_subnets  = [for k, v in local.azs : cidrsubnet(local.cluster_vpc_cidr, 8, k + 48)]
+
+  enable_nat_gateway = true
+  single_nat_gateway = true
+
+  public_subnet_tags = {
+    "kubernetes.io/role/elb" = 1
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/role/internal-elb" = 1
+  }
+
+  tags = local.tags
+}
+
+################################################################################
+# Client VPC
+################################################################################
+
+module "client_vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 5.4"
+
+  name = local.name
+  cidr = local.client_vpc_cidr
+
+  azs             = local.azs
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.client_vpc_cidr, 4, k)]
+
+  tags = local.tags
+}
+
+################################################################################
+# EKS Addons (AWS Gateway API Controller)
+################################################################################
+
+module "addons" {
+  source  = "aws-ia/eks-blueprints-addons/aws"
+  version = "~> 1.12"
+
+  cluster_name      = module.eks.cluster_name
+  cluster_endpoint  = module.eks.cluster_endpoint
+  cluster_version   = module.eks.cluster_version
+  oidc_provider_arn = module.eks.oidc_provider_arn
+
+  enable_aws_gateway_api_controller = true
+  aws_gateway_api_controller = {
+    chart_version           = "v1.0.2"
```

Comment:
please upgrade to v1.0.3, there was a bug fix we published yesterday (serviceImport was not working properly)